weather_test.go

```diff
@@ -36,12 +35,12 @@ func TestGetSendsCorrectURL(t *testing.T) {
 	t.Parallel()
 	var w weather.Conditions
 	s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		data, err := ioutil.ReadFile("testdata/weather.json")
+		file, err := os.Open("testdata/weather.json")
 		if err != nil {
-			fmt.Println("Error: ", err)
-			return
+			t.Fatal(err)
 		}
-		fmt.Fprintln(w, string(data))
+		defer file.Close()
```

Comment:
@bitfield is this required? In the TestParseJson function, we have only done os.open but not done an explicit or a deferred file close.

Reply:
It's always good practice to defer closing the resource.